File: Java/451. Sort Characters By Frequency.java
class Solution {
    public String frequencySort(String s) {
        String str="";
        Map<Character,Integer> mp=new TreeMap<>();
        List<Character> c=new ArrayList<>();
        for(int i=0;i<s.length();i++){
            mp.put(s.charAt(i),mp.getOrDefault(s.charAt(i),0)+1);
            c.add(s.charAt(i));
        }
        
        Collections.sort(c, (a, b) -> {
            int diff = mp.get(b) - mp.get(a);
            if(diff == 0) {
                return b - a;
            }   
            return diff;
        });
        
        
        for(int i=0;i<c.size();i++){
            str=str+c.get(i);
        }
        
        return str;
    }
}

File: C#/451. Sort Characters By Frequency.cs
public class Solution {
    public string FrequencySort(string s) {
        var a = new (int num, char c)[75];

        foreach(var ch in s){
            var (num, c) = a[ch - '0'];
            num++;
            a[ch - '0'] = (num, ch);
        }
        
        Array.Sort(a, (f, s) => s.num - f.num);
        var result = new StringBuilder();
        
        foreach(var (num, c) in a){
            if(num == 0){
                break;
            }

            result.Append(c, num);
        }

        return result.ToString();
    }
}
